src/OpalCompiler-Core/CompiledBlock.extension.st

```diff
@@ -17,6 +17,10 @@ CompiledBlock >> sourceNodeForPC: aPC [
 	blockNode := self outerCode sourceNodeForPC: self pcInOuter.
 	"Bug in the cache? The mapping is returning Return node instead of Block"
 	blockNode isReturn ifTrue: [ blockNode := blockNode value ].
+	self method
+		propertyAt: self printString asSymbol
```

Comment:
this is odd to use the printString as key. Can this not lead to two keys being the same?

Reply:
It could be if a method has two blocks with the exact same code...

I did that, presuming that it should almost never happen and because I have not any better solution. I need to be able to do a mapping between "different versions of compiled blocks"

Reply:
We did have problems with that in the past (e.g. we used to compile just one CompiledBlock if they were the same, but that then meant that the syntax highlight was highlighting wrong. It does not happen often but it does happen.

We have this problem in another place already:
There is now code that searches the compiledBlock with identity, see https://github.com/pharo-project/pharo/issues/11913 . I was thinking that we should encode count blocks an add a unique number e.g. in the trailer.

Reply:
So we could use the simple printString first and improve it late

Reply:
For this case, maybe we could use the PC in outer code as a key. Two blocks cannot be created at the same PC.

If a block is created at PC `48`  in its outer method, its key could be `#48`.
If an embedded block is created inside this block at PC `57`, then its key could be  `#48>57`...

I will try that

Reply:
For this case, maybe we could use the PC in outer code as a key. Two blocks cannot be created at the same PC.

If a block is created at PC `48` in its outer method, its key could be `#48`.
If an embedded block is created inside this block at PC `57`, then its key could be `#48>57`...

I could try that